app/src/main/java/com/arctouch/codechallenge/api/TmdbApi.java

```diff
@@ -16,24 +27,70 @@
     String DEFAULT_LANGUAGE = "pt-BR";
     String DEFAULT_REGION = "BR";
 
+    int TIMEOUT = 60; //Seconds
+    int NUMBER_OF_THREADS = 20;
+
     @GET("genre/movie/list")
-    Observable<GenreResponse> genres(
+    Observable<Response<GenreResponse>> genres(
             @Query("api_key") String apiKey,
             @Query("language") String language
     );
 
     @GET("movie/upcoming")
-    Observable<UpcomingMoviesResponse> upcomingMovies(
+    Observable<Response<UpcomingMoviesResponse>> upcomingMovies(
             @Query("api_key") String apiKey,
             @Query("language") String language,
             @Query("page") Long page,
             @Query("region") String region
     );
 
     @GET("movie/{id}")
-    Observable<Movie> movie(
+    Observable<Response<Movie>> movie(
             @Path("id") Long id,
             @Query("api_key") String apiKey,
             @Query("language") String language
```

Comment:
You're passing `apiKey` and `language` to each method. Can you think of a way to improve this?

Reply:
Yes, there could be added an Interceptor to the OkHttp in which it adds the api_key and language to the HttpUrl each time there is an API call. The code of the Interceptor would look like this:
```
.addInterceptor(chain -> {
    Request request = chain.request();
    HttpUrl url = request.url().newBuilder()
            .addQueryParameter("api_key", TmdbApi.API_KEY)
            .addQueryParameter("language", TmdbApi.DEFAULT_LANGUAGE)
            .build();
    request = request.newBuilder().url(url).build();
    return chain.proceed(request);
})
```